authors/apps/articles/models.py

```diff
@@ -0,0 +1,52 @@
+from django.db import models
+from authors.apps.authentication.models import User
+from django.template.defaultfilters import slugify
+from django.utils import timezone
+import secrets
+
+
+class Article(models.Model):
+
+    slug = models.SlugField(max_length=255)
+
+    title = models.CharField(db_index=True, max_length=255)
+
+    description = models.TextField()
+
+    body = models.TextField()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    updated_at = models.DateTimeField(auto_now_add=True)
+
+    favorited = models.BooleanField(default=False)
+
+    favorites_count = models.IntegerField(default=0)
+
+    author = models.ForeignKey(
+        User,
+        related_name='articles',
+        on_delete=models.CASCADE)
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
+            self.slug = self.get_unique_slag('title', 'slug')
+        super().save(*args, **kwargs)
+
+    def get_unique_slag(self, slugable_field_name, slug_field_name):
+        slug = slugify(self.title)
+        unique_slug = slug
+        # Use a hash string to create a unigue extension to
+        # append to the slug inorder to create a unique slag
+        extention = secrets.token_hex(10)
```

Comment:
This might be an overkill if we're looking at articles with similar slugs.
I would like for you to implement an increase in number instead.
If we have two articles with a similar slug, let the second article's slug have an increment of 1 instead.

This will make it easier when we're looking at the `GET` feature for articles and have to use slugs as identifiers.